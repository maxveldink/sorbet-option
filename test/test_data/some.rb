# typed: true
# frozen_string_literal: true

class TestGenerics
  extend T::Sig

  sig { params(should_be_some: T::Boolean).returns(Typed::Option[Integer]) }
  def test_generic_initializer(should_be_some)
    if should_be_some
      Typed::Some.new(1)
    else
      Typed::None.new
    end
  end

  sig { returns(Typed::Some[Integer]) }
  def test_inferred_payload_type
    some = Typed::Some.new(1)

    T.assert_type!(some.payload, Integer)
    some
  end

  sig { returns(Typed::Some[Integer]) }
  def test_explicit_payload_type
    Typed::Some[Integer].new("success")
  end

  sig { void }
  def test_payload_or
    some = Typed::Some.new("success")

    some.payload_or(1).upcase
  end
end
