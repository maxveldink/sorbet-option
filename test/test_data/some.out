test/test_data/some.rb:26: Expected `Integer` but found `String("success")` for argument `payload` https://srb.help/7002
    26 |    Typed::Some[Integer].new("success")
                                     ^^^^^^^^^
  Expected `Integer` for argument `payload` of method `Typed::Some#initialize`:
    ./lib/typed/option.rb:58:
    58 |    sig { params(payload: Payload).void }
                         ^^^^^^^
  Got `String("success")` originating from:
    test/test_data/some.rb:26:
    26 |    Typed::Some[Integer].new("success")
                                     ^^^^^^^^^

test/test_data/some.rb:33: Method `upcase` does not exist on `Integer` component of `T.any(String, Integer)` https://srb.help/7003
    33 |    some.payload_or(1).upcase
                               ^^^^^^
  Got `T.any(String, Integer)` originating from:
    test/test_data/some.rb:33:
    33 |    some.payload_or(1).upcase
            ^^^^^^^^^^^^^^^^^^
  Did you mean `phase`? Use `-a` to autocorrect
    test/test_data/some.rb:33: Replace with `phase`
    33 |    some.payload_or(1).upcase
                               ^^^^^^
    https://github.com/sorbet/sorbet/tree/19c765c0bbcf09e0e545a50535c6ec0c94690b76/rbi/core/integer.rbi#L921: Defined here
     921 |  def phase(); end
            ^^^^^^^^^^^
